subsys/fs/littlefs_fs.c

```diff
@@ -619,40 +582,123 @@ static lfs_size_t get_block_size(const struct flash_area *fa)
 	return ctx.max_size;
 }
 
-static int littlefs_flash_init(struct fs_mount_t *mountp)
+static int littlefs_init_fmp(struct fs_mount_t *mountp)
 {
 	unsigned int area_id = (uintptr_t)mountp->storage_dev;
 	struct fs_littlefs *fs = mountp->fs_data;
-	const struct flash_area **fap = (const struct flash_area **)&fs->backend;
+	const struct flash_area *area;
 	const struct device *dev;
 	int ret;
 
 	/* Open flash area */
-	ret = flash_area_open(area_id, fap);
-	if ((ret < 0) || (*fap == NULL)) {
+	ret = flash_area_open(area_id, &area);
+	if ((ret < 0) || (area == NULL)) {
 		LOG_ERR("can't open flash area %d", area_id);
 		return -ENODEV;
 	}
 
 	LOG_DBG("FS area %u at 0x%x for %u bytes", area_id,
-		(uint32_t)(*fap)->fa_off, (uint32_t)(*fap)->fa_size);
+		(uint32_t)(area->fa_off), (uint32_t)(area->fa_size));
 
-	dev = flash_area_get_device(*fap);
+	dev = flash_area_get_device(area);
 	if (dev == NULL) {
 		LOG_ERR("can't get flash device: %s",
-			log_strdup((*fap)->fa_dev_name));
+			log_strdup(area->fa_dev_name));
 		return -ENODEV;
 	}
 
-	fs->backend = (void *) *fap;
+	fs->backend = (void *)area;
 	return 0;
 }
 
+/* Block device (blk) backend functions */
+static inline bool lfs_on_blkdev(struct fs_mount_t *mountp)
+{
+	return mountp->flags & FS_MOUNT_FLAG_USE_DISK_ACCESS;
+}
```

Comment:
Shouldn't the `IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV)` be also added (as it was in the removed function - `littlefs_on_blkdev()`)? IIRC @de-nordic was for adding it to this function.